includes/data-stores/class-wc-admin-reports-variations-data-store.php

```diff
@@ -0,0 +1,249 @@
+			$sql_query_params['where_clause'] .= " AND {$wpdb->prefix}posts.post_status IN ( '" . implode( "','", $statuses ) . "' ) ";
+			$sql_query_params['where_clause'] .= ' AND variation_id > 0';
+		}
+
+		return $sql_query_params;
+	}
+
+	/**
+	 * Maps ordering specified by the user to columns in the database/fields in the data.
+	 *
+	 * @param string $order_by Sorting criterion.
+	 * @return string
+	 */
+	protected function normalize_order_by( $order_by ) {
+		if ( 'date' === $order_by ) {
+			return 'date_created';
+		}
+
+		return $order_by;
+	}
+
+	/**
+	 * Enriches the product data with attributes specified by the extended_attributes.
+	 *
+	 * @param array $products_data Product data.
+	 */
+	protected function include_extended_variation_info( &$products_data ) {
+		foreach ( $products_data as $key => $product_data ) {
+			$product             = wc_get_product( $product_data['product_id'] );
+			$extended_attributes = apply_filters( 'woocommerce_rest_reports_products_extended_attributes', $this->extended_attributes, $product_data );
+			foreach ( $extended_attributes as $extended_attribute ) {
+				$function = 'get_' . $extended_attribute;
+				if ( is_callable( array( $product, $function ) ) ) {
+					$value                                        = $product->{$function}();
+					$products_data[ $key ][ $extended_attribute ] = $value;
+				}
+			}
+			$variations = $product->get_available_variations();
+			foreach ( $variations as $variation ) {
+				if ( $variation['variation_id'] == $product_data['variation_id'] ) {
+					$attributes = $variation['attributes'];
+					$formatted_attributes = array();
+					foreach ( $attributes as $att_key => $value ) {
+						$formatted_key = preg_replace( '/(attribute_pa_|attribute_)/', '', $att_key );
+						$formatted_attributes[ $formatted_key ] = $value;
+					}
+					$products_data[ $key ]['attributes'] = $formatted_attributes;
+				}
+			}
+		}
+	}
+
+	/**
+	 * Returns the report data based on parameters supplied by the user.
+	 *
+	 * @param array $query_args  Query parameters.
+	 * @return stdClass|WP_Error Data.
+	 */
+	public function get_data( $query_args ) {
+		global $wpdb;
+
+		$table_name = $wpdb->prefix . self::TABLE_NAME;
+		$now        = time();
+		$week_back  = $now - WEEK_IN_SECONDS;
+
+		// These defaults are only partially applied when used via REST API, as that has its own defaults.
+		$defaults = array(
+			'per_page'              => get_option( 'posts_per_page' ),
+			'page'                  => 1,
+			'order'                 => 'DESC',
+			'orderby'               => 'date',
+			'before'                => date( WC_Admin_Reports_Interval::$iso_datetime_format, $now ),
+			'after'                 => date( WC_Admin_Reports_Interval::$iso_datetime_format, $week_back ),
+			'fields'                => '*',
+			'products'              => array(),
+			'variations'            => array(),
+			'extended_variation_info' => false,
```

Comment:
It's too big and too specific, we probably should extend more data in our endpoints, so probably better change to `extended_data` or `extended_info`, so we can apply it to other endpoints.